triedb/pathdb/disklayer.go

```diff
@@ -225,6 +295,17 @@ func (dl *diskLayer) revert(h *history) (*diskLayer, error) {
 	if dl.id == 0 {
 		return nil, fmt.Errorf("%w: zero state id", errStateUnrecoverable)
 	}
+	var (
+		buff     = crypto.NewKeccakState()
+		accounts = make(map[common.Hash][]byte)
+		storages = make(map[common.Hash]map[common.Hash][]byte)
+	)
+	for addr, blob := range h.accounts {
+		accounts[crypto.HashData(buff, addr.Bytes())] = blob
+	}
+	for addr, storage := range h.storages {
```

Comment:
won't all of these `addr` also be present in the  `accounts`? If so, you've already hashed all of these a few lines above, might want to somehow reuse that?

Reply:
good point!